File: 958-sort-array-by-parity-ii/sort-array-by-parity-ii.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        vector<int>ans(nums.size());
        int i=0;
        int j=1;
        for(int n:nums){
            if(n%2){
                ans[j]=n;
                j+=2;
            }else{
                ans[i]=n;
                i+=2;
            }
        }
        
        return ans;
    }
};

File: 958-sort-array-by-parity-ii/sort-array-by-parity-ii.go
func sortArrayByParityII(nums []int) []int {
    even:=[]int{}
    odd:=[]int{}
    for _,n:=range nums{
        if (n%2==0){
            even=append(even,n)
        }else{
            odd=append(odd,n)
        }
    }  
    ans:=[]int{}
    i:=0
    j:=0
    for i<len(even) && j<len(odd){
        ans=append(ans,even[i])
        ans=append(ans,odd[j])
        i++
        j++
    }  
    return ans
}